plugins/linux-v4l2/v4l2-input.c

```diff
@@ -276,10 +276,12 @@ static void v4l2_device_list(obs_property_t *prop, obs_data_t *settings)
 	const char *cur_device_name;
 
 #ifdef __FreeBSD__
-	dirp = opendir("/dev");
+	const char *basedir = "/dev/";
 #else
-	dirp = opendir("/sys/class/video4linux");
+	const char *basedir = "/dev/v4l/by-path/";
```

Comment:
Can you use `by-id` instead of `by-path`? This will allow plugging capture devices in different USB ports without the need to change the config (the ID will stay the same, but the path will change).